### PR TITLE
Making clusterVersion non-null at all times

### PR DESCRIPTION
**Pull request by tombujok**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java [hunk 1]
```diff
@@ -237,7 +237,7 @@ private void validateNodeCompatibleWith(Version clusterVersion) {
 
     // validate transition from current to newClusterVersion is allowed
     private void validateClusterVersionChange(Version newClusterVersion) {
-        if (clusterVersion != null && clusterVersion.getMajor() != newClusterVersion.getMajor()) {
+        if (clusterVersion != null && !clusterVersion.isUnknown() && clusterVersion.getMajor() != newClusterVersion.getMajor()) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java]: `clusterVersion != null` condition could be removed

Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java]: Thx, fixed!

**Comment by vbekiaris** [comment]
In `ClusterStateChange::validate` cluster version was already guarded against going back to undefined by the `(newState == null)` condition. We should a new condition to ensure that when the cluster state change is of type `ClusterVersion`, requested new cluster version cannot be `UNKNOWN`.

**Comment by tombujok** [comment]
@vbekiaris great finding! I added a fix for that. Could you have a look?

File: hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateChange.java [hunk 2]
```diff
@@ -57,6 +58,12 @@ public void validate() {
             throw new IllegalArgumentException("Invalid null state");
         }
 
+        if (isOfType(Version.class)) {
+            if (((Version) newState).isUnknown()) {
+                throw new IllegalArgumentException("Cannot change Version to UNKNOWN!");
+            }
+        }
+
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateChange.java]: Here's the fix for UNKNOWN version

**Comment by vbekiaris** [comment]
👍 looks good!

**Pull request merged**